src/handlers/glacier.rs

```diff
@@ -0,0 +1,107 @@
+//! Allows team members to directly create a glacier PR with the code provided.
+
+use crate::{
+    config::GlacierConfig,
+    github::Event,
+    handlers::{Context, Handler},
+};
+
+use futures::future::{BoxFuture, FutureExt};
+use parser::command::glacier::GlacierCommand;
+use parser::command::{Command, Input};
+use octocrab::params::repos::Reference;
+use octocrab::models::Object;
+
+pub(super) struct GlacierHandler;
+
+impl Handler for GlacierHandler {
+    type Input = GlacierCommand;
+    type Config = GlacierConfig;
+
+    fn parse_input(
+        &self,
+        ctx: &Context,
+        event: &Event,
+        _: Option<&GlacierConfig>,
+    ) -> Result<Option<Self::Input>, String> {
+        let body = if let Some(b) = event.comment_body() {
+            b
+        } else {
+            // not interested in other events
+            return Ok(None);
+        };
+
+        match event {
+            Event::IssueComment(_) => (),
+            _ => {return Ok(None);}
+        };
+
+        let mut input = Input::new(&body, &ctx.username);
+        match input.parse_command() {
+            Command::Glacier(Ok(command)) => Ok(Some(command)),
+            Command::Glacier(Err(err)) => {
+                return Err(format!(
+                    "Parsing assign command in [comment]({}) failed: {}",
+                    event.html_url().expect("has html url"),
+                    err
+                ));
+            }
+            _ => Ok(None),
+        }
+    }
+
+    fn handle_input<'a>(
+        &self,
+        ctx: &'a Context,
+        _config: &'a GlacierConfig,
+        event: &'a Event,
+        cmd: GlacierCommand,
+    ) -> BoxFuture<'a, anyhow::Result<()>> {
+        handle_input(ctx, event, cmd).boxed()
+    }
+}
+
+async fn handle_input(ctx: &Context, event: &Event, cmd: GlacierCommand) -> anyhow::Result<()> {
+    let is_team_member = if let Err(_) | Ok(false) = event.user().is_team_member(&ctx.github).await
+    {
+        false
+    } else {
+        true
+    };
+
+    if !is_team_member {
+        return Ok(())
+    };
+
+    let url = cmd.source;
+    let response = ctx.github.raw().get(&format!("{}{}", url.replace("github", "githubusercontent"), "/playground.rs")).send().await?;
+    let body = response.text().await?;
+
+    let number = event.issue().unwrap().number;
+    let user = event.user();
+
+    let octocrab = &ctx.octocrab;
+
+    let fork = octocrab.repos("rustbot", "glacier");
+    let base = octocrab.repos("rust-lang", "glacier");
+
+    let master = base.get_ref(&Reference::Branch("master".to_string())).await?.object;
+    let master = if let Object::Commit { sha, ..} = master {
+        sha
+    } else {
+        panic!()
```

Comment:
Please add the actual value of master here so that we can debug if it does occur.

Reply:
Well I think this should be `unreachable!()` instead.

Reply:
Regardless, we should note what we actually encountered.